### PR TITLE
Fix gulpfile tasks in release zip, but limit packaged scripts

### PR DESCRIPTION
**Pull request by ggetz**
Fixes https://github.com/CesiumGS/cesium/issues/3911

We removed the `gulpfile` in https://github.com/CesiumGS/cesium/pull/10281, but after seeing #6025, I think that was the wrong move.

The reason we removed it was that it was no longer possible to run _any_ gulp tasks from within the release zip. However, this was due to https://github.com/CesiumGS/cesium/pull/8659 is fixed here by including `.gulp.json`. 

Additionally, it's only possible and helpful a subset of scripts in the release zip. So this PR limits those to the following list:

* `start`
* `start-public`
* `clean`
* `cloc`
* `coverage`
* `eslint`
* `build-specs`
* `test-*`
* `prettier-check`

This list is arrived at by omitting build and watch tasks, as well as the `deploy-*` tasks. Including `generateDocumentation` and `build-ts` is arguable since they can be run to verify the validity of the docs. But we already included the generated output, and the size of the documentation build tools are not as negligable as eslint or prettier. 

In order to run those, we include include `.eslintrc.json`,`.eslintignore`, related eslint config files in the release zip (~7KB uncompressed); `.prettierignore` (<1KB uncompressed); and `Specs/**` which was previously, but I believe erroneously, excluded in https://github.com/CesiumGS/cesium/pull/10281.

Finally, after https://github.com/CesiumGS/cesium/pull/10293 adds back the browser spec runner and associated files, we can re-add the link to the spec runner removed from `index.release.html` in https://github.com/CesiumGS/cesium/pull/10184.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/packaged-scripts/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: gulpfile.cjs [hunk 1]
```diff
@@ -451,6 +451,29 @@ gulp.task(
...
+    delete packageJson.scripts.buildApps;
+    delete packageJson.scripts.combine;
+    delete packageJson.scripts.combineRelease;
+    delete packageJson.scripts.generateDocumentation;
+    delete packageJson.scripts["generateDocumentation-watch"];
+    delete packageJson.scripts.makeZipFile;
+    delete packageJson.scripts.minify;
+    delete packageJson.scripts.minifyRelease;
+    delete packageJson.scripts.release;
+    delete packageJson.scripts.combineRelease;
```
Comment (ebogo1) [inline gulpfile.cjs:469]: duplicate of line 462?

**Review by ebogo1 — COMMENTED** [review]
Looks good, just a minor comment!

File: gulpfile.cjs (hunk 1 above)
Comment (ggetz) [reply gulpfile.cjs]: Yes, good catch!

**Comment by ggetz** [comment]
Thanks @ebogo1! Updated.

**Comment by ebogo1** [comment]
Hm, this is a pretty niche use case but I noticed that if we run `npm run clean` followed by `npm run build-specs`, the latter fails because `clean` removes the SpecList.js file. This happens in the non-release version as well, but if there's no way to rebuild SpecList.js from the release zip (it's part of the `"build"` task) maybe it's worth reconsidering how we include the `build-specs` task. 

I think it's minor enough to not do anything about it in this PR but thought I'd mention it.

**Comment by ggetz** [comment]
Good point. Maybe we should omit the `clean` script entirely then, since its possible to clean files which can't be re-created from within the release zip.

**Comment by ebogo1** [comment]
Should the `cloc` script be removed too then?

**Comment by ggetz** [comment]
Yeah, I think that follows. I also don't see an equivalent command in packages from comparable projects.

**Comment by ggetz** [comment]
Updated once again.

**Comment by ebogo1** [comment]
Cool, thanks @ggetz!

**Pull request merged**